### PR TITLE
Stage 4 Remaster

### PR DESCRIPTION
**Pull request by laochailan**
This port is more or less complete now.

The only real leftovers are the visuals of the adds in `vlads_army`. Should they get their own “enemy class” just for this one spell?

### CODE REVIEW

File: src/stages/stage4/timeline.c [hunk 1]
```diff
@@ -18,116 +18,85 @@
...
+			.proto = rng_chance(0.5) ? pp_rice : pp_thickrice,
 			.pos = e->pos,
-			.color = RGB(0.8,0.3-0.1*afrand(1),0.5),
-			.rule = accelerated,
-			.args = {
-				e->args[0]/2+(1-2*afrand(2))+(1-2*afrand(3))*I,
-				0.02*I
-			}
+			.color = RGB(0.8, 0.3 - 0.1 * vrng_real(rand[0]), 0.5),
+            .move = move_accelerated(move / 2 + (1 - 2 * vrng_real(rand[1])) + (1 - 2 * vrng_real(rand[2])) * I, 0.02 * I),
```
Comment (StarWitch) [inline src/stages/stage4/timeline.c:52]: Indentation looks wrong here.

File: src/stages/stage4/timeline.c [hunk 2]
```diff
@@ -503,143 +443,188 @@ static int scythe_post_mid(Enemy *e, int t) {
...
+	for(int i = 0; i < 40; i++) {
+		cmplx pos = VIEWPORT_W * (i & 1) + I * (20.0 + VIEWPORT_H/3.0 * rng_real());
+		cmplx vel = 3 * (1 - 2 * (i & 1)) + I;
+		INVOKE_TASK_DELAYED(1800 + 10 * i, explosive_swirl, .pos = pos, .move = move_linear(vel));
 	}
+
+	WAIT(2300);
+	INVOKE_TASK(spawn_boss);
+	WAIT_EVENT(&NOT_NULL(global.boss)->events.defeated);
+	stage_unlock_bgm("stage2boss");
```
Comment (StarWitch) [inline src/stages/stage4/timeline.c:621]: `stage4boss`

File: src/stages/stage4/kurumi.c [hunk 3]
```diff
@@ -14,6 +14,34 @@
 
 MODERNIZE_THIS_FILE_AND_REMOVE_ME
```
Comment (StarWitch) [inline src/stages/stage4/kurumi.c:15]: Don't need this anymore.

**Review by StarWitch — DISMISSED** [review]
This runs correctly, no crashes! I think it's good to go.

**Comment by Akaricchi** [comment]
Merged as 93c68dbb0d1a822fb61e9783426900db639ae5db

**Pull request closed without merging**